sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/DirectoryApiTests.java

```diff
@@ -125,6 +125,25 @@ public void getFileClient() {
         assertInstanceOf(ShareFileClient.class, fileClient);
     }
 
+    private static Stream<Arguments> getNonEncodedFileNameSupplier() {
+        return Stream.of(
+            Arguments.of("test%test"),
+            Arguments.of("%Россия 한국 中国!"),
+            Arguments.of("%E6%96%91%E9%BB%9E"),
+            Arguments.of("斑點")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("getNonEncodedFileNameSupplier")
+    public void getNonEncodedFileName(String fileName) {
+        primaryDirectoryClient.create();
+        ShareFileClient fileClient = primaryDirectoryClient.getFileClient(fileName);
+        assertEquals(primaryDirectoryClient.getDirectoryPath() + "/" + fileName, fileClient.getFilePath());
```

Comment:
Shouldn't the file name be encoded when it is part of the URL?

Reply:
The URL does not need to contain the file name pre-encoded because when the specific API makes a service call, in the implementation methods there is a specific parameter called @PathParam that will handle encoding (through azure-core) if the encoded flag is not passed. For example, when calling fileClient.create(), it calls into this method:
https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/implementation/FilesImpl.java#L95

and will handle encoding the file names and urls appropriately through azure-core.